### PR TITLE
Added TraceIdStore implementation - to protect against potential fraud threats

### PR DESCRIPTION
**Pull request by harikareddy1023**

### CODE REVIEW

File: src/main/java/org/piangles/gateway/requests/CacheTraceIdStore.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import org.piangles.core.util.InMemoryConfigProvider;
+
+public class CacheTraceIdStore implements TraceIdStore 
+{
+	private RedisCache redisCache = null;
+
+	public CacheTraceIdStore() 
+	{
+		redisCache = ResourceManager.getInstance().getRedisCache(new InMemoryConfigProvider(, "c6a1426f-34b9-4f6f-9ff6-bb0c7f200c5b"));
+	}
```
Comment (harikareddy1023) [inline src/main/java/org/piangles/gateway/requests/CacheTraceIdStore.java:14]: TODO where does the componentId go for Gateway Config

File: src/main/java/org/piangles/gateway/GatewayConfiguration.java [hunk 2]
```diff
@@ -55,4 +57,9 @@ public int getMaxTextMessageSize()
 	{
 		return maxTextMessageSize;
 	}
+
+	public boolean getCacheTraceIdStoreEnabled()
```
Comment (saradhiZuro) [inline src/main/java/org/piangles/gateway/GatewayConfiguration.java:61]: Please rename getCacheTraceIdStoreEnabled to isCacheTraceIdStoreEnabled

File: src/main/java/org/piangles/gateway/requests/RequestProcessingManager.java [hunk 3]
```diff
@@ -240,6 +243,44 @@ public void onMessage(String messageAsString)
...
+				traceIdStore = new InMemoryTraceIdStore();
+			}
+
+			//check if the traceId is present in Redis cache
+			boolean found = traceIdStore.exists(traceId);
+			if (found)
+			{
+				logger.warn("TraceId: " + request.getTraceId() + "is being reused, FraudAction detected");
+				//un-reqister the session
+				sessionService.unregister(clientDetails.getSessionDetails().getUserId(), clientDetails.getSessionDetails().getSessionId());
```
Comment (saradhiZuro) [inline src/main/java/org/piangles/gateway/requests/RequestProcessingManager.java:269]: We need to close connection.

File: src/main/java/org/piangles/gateway/requests/CacheTraceIdStore.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+		});
+	}
+
+	@Override
+	public boolean exists(String traceId) throws Exception
+	{
+		String found = redisCache.execute((jedis) -> jedis.get(traceId));
+
+		return found != null;
+	}
```
Comment (saradhiZuro) [inline src/main/java/org/piangles/gateway/requests/CacheTraceIdStore.java:34]: We need a private method to createKey(String traceId) {return "TraceId:" + traceId;}

**Review by saradhiZuro — CHANGES_REQUESTED** [review]

**Review by saradhiZuro — APPROVED** [review]

**Pull request merged**